chevah/python-package
# [#4974] Test for outdated and insecure Python modules with pip and safety.

Scope
=====

We might want to look into setting up something like ​​https://pyup.io/safety/

Is free for on-premise deployment.

Changes
=======

Use `pip list --outdated` to show old Python modules.

Use `safety` to check for known vulnerabilities in Python modules and error out if any found. 
But do not bundle `safety`, only install it with `pip` during test phase.

**Drive-by fixes**:
  * updated `pip` and `get-pip.py` to latest version: 18.0
  * also copy `paver.conf` alongside `paver.sh` for compat tests, to avoid incompatibilities
  * updated `pypiwin32` and renamed accordingly to upstream to `pywin32`, added test for it
  * updated `scandir` to 1.9.0 (this fixes testing in OpenBSD)
  * install some modules through `pip` even on Solaris 10 and HP-UX using `PIP_LIBRARIES_NO_CFFI`
  * adjusted list of required packages for SLES 12 to account for the SP3 upgrade.

How to try and test the changes
===============================

reviewers: @adiroiban 

Please double-check changes in `chevah_build` and `python-modules/chevah-python-test/test_python_binary_dist.py`

Run the automated tests and check for the output of our tests.
`pip` should list **all** outdated packages, eg. pyOpenSSL everywhere, as 18.0.0 was released and we are at most at 17.5.0.
`safety` should list **all** packages with known vulnerabilities.

Make sure the test phase errors out if any vulnerabilities are found.
Note that for some reason `safety` doesn't error on Windows (and only on Windows), eg. https://chevah.com/buildbot/builders/python-package-win-xp-dev/builds/62/steps/test/logs/stdio. Reported upstream at https://github.com/pyupio/safety/issues/153.
